ios/RNCWKWebView.m

```diff
@@ -88,7 +88,22 @@ - (void)didMoveToWindow
         wkWebViewConfig.processPool = [[RNCWKProcessPoolManager sharedManager] sharedProcessPool];
     }
     wkWebViewConfig.userContentController = [WKUserContentController new];
-    [wkWebViewConfig.userContentController addScriptMessageHandler: self name: MessageHanderName];
+
+    if (_messagingEnabled) {
+      [wkWebViewConfig.userContentController addScriptMessageHandler:self name:MessageHandlerName];
+
+      NSString *source = [NSString stringWithFormat:
+        @"window.%@ = {"
```

Comment:
Would it not be better/more consistent to do:

window.%@ = window.webkit.messageHandlers.%@

Reply:
This is definitely not the same as it is now; if window.webkit.messageHandlers.%@ is not defined when this code runs it won't work, but if we know it always will be then the code I suggested will be more in line with the Android implementation.

Reply:
Thinking of this some more; what if people decide to extend the window.ReactNativeWebview object; is there any chance their code might run before the code here is run (same question for Android)?

Should we check whether this object exists and if so only add the postMessage method? 

I don't think that this requires any changes at this point, but I'm just considering scenarios where this might cause issues. 

If this is a problem, then a simple fix would obviously be for end users not to extend the object and create their own.